plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java

```diff
@@ -327,8 +327,11 @@ public void stop(Throwable t) {
     }
 
     public List<StackTraceElement> getStackTrace() {
-        assert program != null; // Otherwise this CpsThread is not even part of the CpsThreadGroup, so how is it being accessed?
```

Comment:
Looks like the answer is that this can happen if the thread group is modified concurrently since `CpsFlowExecution.getThreadDump` does not use `runInCpsVmThread` and the lower-level methods do not do anything to synchronize the thread dump creation operation with respect to thread group modifications.